lib/invoker.js

```diff
@@ -3,20 +3,28 @@
  */
 const { CloudEvent, HTTP } = require('cloudevents');
 
-module.exports = function invoker(func) {
+module.exports = function invoker(opts) {
+  const func = opts.func;
   return async function invokeFunction(context, log) {
     // Default payload values
     const payload = {
       code: 200,
       response: undefined,
       headers: {
         'content-type': 'application/json; charset=utf-8',
-        'access-control-allow-methods':
-          'OPTIONS, GET, DELETE, POST, PUT, HEAD, PATCH',
-        'access-control-allow-origin': '*'
+        'access-control-allow-methods': 'OPTIONS, GET, POST',
```

Comment:
why are you removing the other methods(DELETE, PUT, HEAD, PATCH)?  This should stay the same since it is the default, otherwise it would be a breaking change

Reply:
**invocation-handler.js**

```javascript
module.exports = function use(fastify, opts, done) {
  fastify.get('/', doGet);
  fastify.post('/', doPost);
  fastify.options('/', doOptions);
  const invokeFunction = invoker(opts);
```

As you can see here, only the post and get were supported and now i added options. Why should the rest of the methods be allowed if they cannot be used? Anyway, I added the rest of allowed methods in cors.